core/src/main/java/org/keycloak/crypto/Algorithm.java

```diff
@@ -33,6 +33,8 @@ public interface Algorithm {
     String PS256 = "PS256";
     String PS384 = "PS384";
     String PS512 = "PS512";
+    String Ed25519 = "Ed25519";
+    String Ed448 = "Ed448";
```

Comment:
This is still happening, don't know if you are still working on this but just in case I didn't explain OK. With the current code if I get the received access token from [this line](https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/AccessTokenTest.java#L1369) when executing the test `AccessTokenTest#accessTokenRequest_ClientEd25519_RealmES256`. I received the following token:

```
eyJhbGciOiJFZDI1NTE5IiwidHlwIiA6ICJKV1QiLCJraWQiIDogImg4X01leWVXOUs4ZUw3NUNxSW9GVGFyV0JCVTdIZXlxcTlHenR5SVlweE0ifQ.eyJleHAiOjE2OTg0MDYxNTUsImlhdCI6MTY5ODQwNTg1NSwiYXV0aF90aW1lIjoxNjk4NDA1ODU1LCJqdGkiOiIyYzM0MmJjOC0yM2FkLTRmOWQtODI2My1jYzA0NTFkMDcxOTAiLCJpc3MiOiJodHRwczovL2xvY2FsaG9zdDo4NTQzL2F1dGgvcmVhbG1zL3Rlc3QiLCJzdWIiOiJjMTg2M2U1Yy02ZjZjLTQ5YzAtODdkZi00ODU2MGI4MDdiMjAiLCJ0eXAiOiJCZWFyZXIiLCJhenAiOiJ0ZXN0LWFwcCIsInNlc3Npb25fc3RhdGUiOiIzYjEyYTlmZi0wYzFjLTRiNGEtOWUwOC1kMmRkMjIwYTc0MmYiLCJhY3IiOiIxIiwiYWxsb3dlZC1vcmlnaW5zIjpbImh0dHA6Ly9sb2NhbGhvc3Q6ODE4MCIsImh0dHBzOi8vbG9jYWxob3N0Ojg1NDMiXSwicmVhbG1fYWNjZXNzIjp7InJvbGVzIjpbInVzZXIiXX0sInJlc291cmNlX2FjY2VzcyI6eyJ0ZXN0LWFwcCI6eyJyb2xlcyI6WyJjdXN0b21lci11c2VyIl19fSwic2NvcGUiOiJvcGVuaWQgcHJvZmlsZSBlbWFpbCIsInNpZCI6IjNiMTJhOWZmLTBjMWMtNGI0YS05ZTA4LWQyZGQyMjBhNzQyZiIsImVtYWlsX3ZlcmlmaWVkIjpmYWxzZSwibmFtZSI6IlRvbSBCcmFkeSIsInByZWZlcnJlZF91c2VybmFtZSI6InRlc3QtdXNlckBsb2NhbGhvc3QiLCJnaXZlbl9uYW1lIjoiVG9tIiwiZmFtaWx5X25hbWUiOiJCcmFkeSIsImVtYWlsIjoidGVzdC11c2VyQGxvY2FsaG9zdCJ9.B2UbYvMudUHAyXXNq0mS1qstj4O5mCRQjR5uuVrp5TMF2hD-OZm_cukhFbQK3hSvTVHebCh_QOc6XndANo93Dw
```

And http://jwt.io the headers is:

```json
{
  "alg": "Ed25519",
  "typ": "JWT",
  "kid": "h8_MeyeW9K8eL75CqIoFTarWBBU7Heyqq9GztyIYpxM"
}
```

And I think that "alg" should be `EdDSA` and not `Ed25519`. If I'm not wrong after reading the spec both curves should use `EdDSA` and later the curve is selcted depending the key. Maybe I'm missing something, so please double check. :wink:

Reply:
Yes, I will work for fixing it ("alg": "EdDSA", "crv": "Ed25519" or "Ed448").

Reply:
I still don't get this. The [rfc8037](https://www.rfc-editor.org/rfc/rfc8037.html#section-3) in the signature section says that:

> The EdDSA variant used is determined by the subtype of the key
   (Ed25519 for "Ed25519" and Ed448 for "Ed448").

So I think the header should not contain the `crv` claim. IMHO the curve is just determined by the key, so once you get the key you know if it's one curve or the other. WDYT? Am I understanding the spec wrong?

One option is defining only one algorithm `EdDSA` at keycloak server (which can be one or the other curve) and once the key is got you can know the real curve. I know it changes the PR and probably `crv` is needed at KeyWrapper (extra property or something), but I fear that the RFC forces this way.